dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/util/SqlUtils.java

```diff
@@ -133,13 +151,28 @@ public static Set<String> getArrayAsSet(ResultSet rs, String columnLabel) throws
     return Sets.newHashSet(array);
   }
 
+  /**
+   * Cast the given expression to a target type like {@code cast (X as integer)}.
+   *
+   * <p>See {@link <a
+   * href="https://www.postgresql.org/docs/current/sql-expressions.html#SQL-SYNTAX-TYPE-CASTS">PostgreSQL
+   * type casts</a>}
+   *
+   * @param expression the expression to cast
+   * @param type the target type to cast the expression to
+   * @return the type cast expression string
+   */
+  public static String cast(String expression, String type) {
+    return "cast (" + expression + " as " + type + ")";
+  }
+
   /**
    * Cast the given value to numeric (cast(X as numeric).
    *
    * @param value the value.
    * @return a string with the numeric cast statement.
    */
-  public static String castToNumber(String value) {
+  public static String castToNumeric(String value) {
     return "cast (" + value + " as numeric)";
```

Comment:
Nitpicky, but couldn't this reuse the cast(String expression, String type) method by passing on numeric into it?

Reply:
I'll do that in the next PR